timemachine/fe/atom_mapping.py

```diff
@@ -236,6 +243,78 @@ def _deduplicate_all_cores(all_cores):
     return list(unique_cores.values())
 
 
+def ring_forming_breaking_count(mol_a, mol_b, core: NDArray) -> int:
+    """Counts the number of rings broken and formed given a core. Does not count ring insertion/deletions. Considers
+    the number number of rings broken/formed in both endstates, reversing the mol order and the core order produces
+    the same value.
+
+    The algorithm is to add all ring atoms/bonds to a graph, and count the cycles that contain both core and mol a or
+    mol b atoms.
+    """
+    g = nx.Graph()
+    core_a_to_b = {a: b for a, b in core}
```

Comment:
nit: it seems like we only look at the keys of this dict. Simpler to define e.g. `core_atoms_a` as a set?

Reply:
Converted to sets in https://github.com/proteneer/timemachine/pull/1363/commits/dc38731f87e365add2183c2c2b91fffd914bb896